### ISSUE TITLE
🐞 CS-Flow Model initializes EfficientNet-B5 with NaN BatchNorm running_var

### ISSUE THREAD

**Issue by monzelr** [open-edge-platform/anomalib#2965]
### Describe the bug

When I initialize the CSFLowModel and try to train, I always got NaN values returned after forwarding. With this NaN values in the first train epoch it is not possible to calucalte the loss values and thus the whole training fails.

 I found out that this comes from some layers from the FeatureExtraction which have NaNs inside:

```python
import torch
import torchvision
import sys
import platform
from anomalib.models.image.csflow.torch_model import CsFlowModel

info = {
    "python": sys.version.replace("\n", " "),
    "platform": platform.platform(),
    "torch": torch.__version__,
    "torchvision": torchvision.__version__,
    "cuda_available": str(torch.cuda.is_available()),
    "cuda_version": torch.version.cuda or "None",
    "cudnn_version": str(torch.backends.cudnn.version()) if torch.backends.cudnn.is_available() else "None",
    "gpu_name": torch.cuda.get_device_name(0) if torch.cuda.is_available() else "None",
    "compute_capability": str(torch.cuda.get_device_capability(0)) if torch.cuda.is_available() else "None",
    "allow_tf32_matmul": str(getattr(torch.backends.cuda.matmul, "allow_tf32", None)),
    "allow_tf32_cudnn": str(getattr(torch.backends.cudnn, "allow_tf32", None)),
}
for key, value in info.items():
    print(key, value)

model = CsFlowModel(
    input_size=(640, 1280),
    cross_conv_hidden_channels=1024,
    n_coupling_blocks=4,
    clamp=3,
    num_channels=3,
).eval()

print("\n# Probe BN in backbone: features.0.1")
bn = model.feature_extractor.feature_extractor.backbone.features[0][1]  # BatchNorm2d
def s(t): return (torch.isfinite(t).all().item(),
                  float(t.min().item()), float(t.max().item()), float(t.mean().item()))
print("affine weight finite:", torch.isfinite(bn.weight).all().item())
print("affine bias   finite:", torch.isfinite(bn.bias).all().item())
print("running_mean  stats :", s(bn.running_mean))
print("running_var   stats :", s(bn.running_var))
```

My output:
```
# Probe BN in backbone: features.0.1
affine weight finite: True
affine bias   finite: True
running_mean  stats : (False, nan, nan, nan)
running_var   stats : (False, nan, nan, nan)
```

Can anyone confirm this? Do I miss something?

### Dataset

N/A

### Model

CSFlow

### Steps to reproduce the behavior

see script above

### OS information

python 3.10.12 (main, Aug 15 2025, 14:32:43) [GCC 11.4.0]
platform Linux-5.15.148-tegra-aarch64-with-glibc2.35
torch 2.8.0
torchvision 0.23.0
cuda_available True
cuda_version 12.6
cudnn_version 90300
gpu_name Orin
compute_capability (8, 7)
allow_tf32_matmul False
allow_tf32_cudnn True

### Expected behavior

BatchNorm buffers (`running_mean`, `running_var`) of EfficientNet-B5 are finite right after `CsFlowModel` construction.

### Pip/GitHub

pip

### What version/branch did you use?

v2.1

### Configuration YAML

```yaml
no configuration yaml file needed
```

### Logs

```shell
see above
```

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

**Comment by monzelr** [comment]
I have found the issue:
When the backbone of CSFlow is generated with the class `TimmFeatureExtractor`, it does a dry run in training-mode and thus modifies the weights of the backbone. You can see this in [line 115 of timm.py](https://github.com/open-edge-platform/anomalib/blob/5f89bbefec128578a5e027fecf7126efa4766a00/src/anomalib/models/components/feature_extractors/timm.py#L115).

However, the backbone shall never be modified in any case. Thus, I suggest updating `dryrun_find_featuremap_dims` in `utils.py`:


```python
def dryrun_find_featuremap_dims(
    feature_extractor: GraphModule,
    input_size: tuple[int, int],
    layers: list[str],
) -> dict[str, dict[str, int | tuple[int, int]]]:
    """Get feature map dimensions by running an empty tensor through the model, safely."""
    device = next(feature_extractor.parameters()).device
    dryrun_input = torch.empty(1, 3, *input_size).to(device)

    # Save current mode
    was_training = feature_extractor.training
    feature_extractor.eval()

    with torch.no_grad():
        dryrun_features = feature_extractor(dryrun_input)

    # Restore previous mode
    if was_training:
        feature_extractor.train()

    return {
        layer: {
            "num_features": dryrun_features[layer].shape[1],
            "resolution": dryrun_features[layer].shape[2:],
        }
        for layer in layers
    }
```
